jumpscale/sals/nettools/__init__.py

```diff
@@ -1,148 +1,279 @@
-"""
-Docs for nettools
+"""Docs for nettools
 
-"""
-# TODO: test *2
+TODO
 
+General Note on python socket operations:
+    If you use a hostname in the host portion of IPv4/v6 socket address,the program may show a nondeterministic behavior,
+    as Python uses the first address returned from the DNS resolution. The socket address will be resolved differently
+    into an actual IPv4/v6 address, depending on the results from DNS resolution and/or the host configuration.
+    For deterministic behavior use a numeric address in host portion.
+    https://docs.python.org/3/library/socket.html
+"""
 import time
-import urllib3
 import socket
 import ipaddress
 import re
 from typing import Optional
-from jumpscale.core.exceptions import Value, Runtime, Input
-from jumpscale.data.time import now
+from jumpscale.core.exceptions import Value, Runtime
 import jumpscale.tools.http
 import jumpscale.data.platform
 import jumpscale.sals.fs
 import jumpscale.core.executors
-from jumpscale.data.types import IPAddress
-
+from urllib.request import Request, urlopen
+from urllib.error import URLError, HTTPError
+import ssl
+import json
+import subprocess
+from urllib.parse import urlparse
+from pathlib import Path
+import shutil
+from urllib.request import build_opener, HTTPPasswordMgrWithDefaultRealm, HTTPBasicAuthHandler, install_opener
+from collections import namedtuple
+from os.path import basename
+from jumpscale.loader import j
+from itertools import count
+
+# going to remove j.data.types . use insted jumpscale.core.base.fields.IPAddress
```

Comment:
> You can cherry pick this commit if possible [bbb2d0f](https://github.com/threefoldtech/js-ng/commit/bbb2d0fc2ecd5e1fe6616a2a343312ca0fb5453d)

jumpscale.core.base.fields.IPAddress not used atm.
i left the memo in case we need to use it later in this module.